client/src/components/Ticketing/ticketingmanager/SubscriptionRedemption/SubscriptionRedemptionContainer.tsx

```diff
@@ -0,0 +1,245 @@
+  setPopUpProps: any;
+}
+
+export const SubscriptionRedemptionContainer = (
+  props: SubscriptionContainerProps,
+) => {
+  const {
+    subscription,
+    setSubscriptions,
+    setDisabled,
+    disabled,
+    token,
+    setReload,
+    showPopUp,
+    setPopUpProps,
+  } = props;
+  const [formStatus, setFormStatus] = useState<
+    'loading' | 'redeeming' | 'nothing'
+  >('nothing');
+  const [submittingStatus, setSubmittingStatus] = useState<
+    'invalid' | 'submitting' | 'nothing'
+  >('nothing');
+  const [eventRestriction, setEventRestriction] = useState(true);
+  const [eventRestrictionIsViolated, setEventRestrictionIsViolated] =
+    useState(false);
+  const [delay, setDelay] = useState(null);
+  const formRef = useRef<FormikProps<FormikValues>>(null);
+  const stateRef = useRef(formStatus);
+  const isMounted = useIsMounted();
+  stateRef.current = formStatus;
+
+  useTimeout(() => {
+    setDelay(null);
+    if (stateRef.current !== 'loading') return;
+    setFormStatus('nothing');
+    setDisabled(false);
+    setPopUpProps(
+      'Failed to Load Subscription',
+      'Error loading subscription please try again',
+      false,
+      'subscription-loading-error-pop-up',
+    );
+  }, delay);
+
+  const onSubmitSuccess = async (event) => {
+    try {
+      const data = await event.json();
+      setSubscriptions((prev) =>
+        prev.map((sub) => (sub.id === data.id ? data : sub)),
+      );
+    } catch (error) {
+      setReload((prev) => !prev);
+    } finally {
+      if (isMounted()) setFormStatus('nothing');
+      setDisabled(false);
+      setPopUpProps(
+        'Save Success',
+        'Successfully updated subscription redemption',
+        true,
+        `subscription-redemption-${subscription.id}-success`,
+      );
+    }
+  };
+
+  const onSubmitError = async (event) => {
+    setPopUpProps(
+      'Save Failure',
+      event.json
+        ? (await event.json()).error
+        : 'Error updating subscription redemption',
+      false,
+      `subscription-redemption-${subscription.id}-failure`,
+    );
+    setFormStatus('nothing');
+    setDisabled(false);
+  };
+
+  return (
+    <article className='bg-white rounded-xl p-4 shadow-xl w-[100%] min-[1440px]:w-[80%]'>
+      <header className='text-zinc-800 text-center flex flex-col min-[768px]:flex-row min-[768px]:justify-between'>
+        <h2 className='flex flex-col mb-2 min-[768px]:mb-0'>
+          <span className='flex flex-col min-[375px]:flex-row max-[767px]:justify-center'>
+            <span>
+              {subscription.firstname} {subscription.lastname}
+            </span>
+            <span className='max-[374px]:hidden px-1'>-</span>
+            <span className='italic'>{subscription.email}</span>
+          </span>
+          <span className='flex flex-col min-[375px]:flex-row max-[767px]:justify-center'>
+            <span>{subscription.name} Subscription</span>
+            <span className='max-[374px]:hidden px-1'>-</span>
```

Comment:
I like the attention to detail, although we don't typically need to worry about screen sizes less than 390px. Particularly on the admin side of things.